### PR TITLE
[WFNC-86] Remove deprecated InitialContext

### PR DESCRIPTION
**Pull request by tadamski**
https://issues.redhat.com/browse/WFNC-86

### CODE REVIEW

File: src/main/java/org/wildfly/naming/client/ProviderEnvironment.java [hunk 1]
```diff
@@ -293,52 +290,21 @@ public Builder populateFromEnvironment(Map<String, ?> environment) throws Naming
...
-                    throw Messages.log.failedToLoadCallbackHandlerClass(e, callbackClass);
-                } catch (Exception e) {
-                    throw Messages.log.failedToInstantiateCallbackHandlerInstance(e, callbackClass);
-                }
-            }
+            boolean overrideDefaultAuth = credentials != null || userName != null;
 
             OptionMap remotingOptions = getOptionMap(environment, CONNECT_OPTIONS_PREFIX, Builder.class.getClassLoader());
 
-            if (callbackHandler != null || userName != null) {
```
Comment (ropalka) [inline src/main/java/org/wildfly/naming/client/ProviderEnvironment.java:322]: Only _callbackHandler != null_ check should be removed from this if condition and not whole _if_ block IMHO.

**Review by ropalka — CHANGES_REQUESTED** [review]

File: src/main/java/org/wildfly/naming/client/ProviderEnvironment.java (hunk 1 above)
Comment (asoldano) [reply src/main/java/org/wildfly/naming/client/ProviderEnvironment.java:322]: Do we have a test covering this? If now, perhaps we should.

Comment (tadamski) [reply src/main/java/org/wildfly/naming/client/ProviderEnvironment.java:322]: I have fixed the mentioned issue and added the test that checks whether global env principal and credentials are propagated to the AuthenticationConfiguration.

**Comment by ropalka** [comment]
This PR is related to https://github.com/wildfly/wildfly/pull/16447

**Review by ropalka — APPROVED** [review]

**Comment by ropalka** [comment]
This PR is related to https://github.com/wildfly/wildfly-arquillian/pull/277

**Comment by ropalka** [comment]
Thanks @tadamski !

**Pull request merged**